pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -355,6 +359,15 @@ func ConfigFromEnv() Config {
 	cfg.Global.ApplicationCredentialName = os.Getenv("OS_APPLICATION_CREDENTIAL_NAME")
 	cfg.Global.ApplicationCredentialSecret = os.Getenv("OS_APPLICATION_CREDENTIAL_SECRET")
 
+	endpointType := os.Getenv("OS_ENDPOINT_TYPE")
+	if endpointType == "internal" || endpointType == "internalURL" {
```

Comment:
You have documented supported types are public, admin,internal, So I dont think its necessary to check the URL also?

Reply:
OpenStack python clients use `URL` suffix, gophercloud package doesn't use it. Extra options are needed to keep the compatibility.